File: matlab/AggressiveDriver.m
classdef AggressiveDriver < Driver
    
    methods
        
        function this = AggressiveDriver(name,weight) 
            this = this@Driver(name, weight);
        end
        
        function [x] = throttle_action(this, car,  env)
                 if(car.current_speed<car.max_speed)
                   x = 0.5*car.max_acceleration;
                 else
                   x = -0.5*car.max_acceleration;
                 
                 end
        end
        
            
        function [x] = stop_for_refueling(this,  car,  env)
            x = false;
        end
        
    end
    
end

File: java/src/AggressiveDriver.java
public class AggressiveDriver extends Driver {

    public AggressiveDriver(String name, double weight) {
        super(name, weight);
    }

    @Override
    public double throttle_action(Car car, Environment env) {

        if(car.current_speed<car.max_speed)
            return 0.5*car.max_acceleration;
        else
            return -0.5*car.max_acceleration;
    }

    @Override
    public boolean stop_for_refueling(Car car, Environment env) {
        return false;
    }

}
